networkit/cpp/generators/BarabasiAlbertGenerator.cpp

```diff
@@ -84,20 +84,18 @@ Graph BarabasiAlbertGenerator::generateOriginal() {
         while (targets.size() - 1 < k) {
             uint64_t random = (uint64_t) Aux::Random::integer(degreeSum);
```

Comment:
`Aux::Random::integer` creates a new integer distribution every time. It is more efficient to create the generator and the random distribution before (as you already did in the faster function below). I think it would be nice to have this optimization also here.